src/core/dev/ring_simple.cpp

```diff
@@ -472,29 +472,28 @@ int ring_simple::socketxtreme_poll(struct xlio_socketxtreme_completion_t *xlio_c
                     xlio_completions++;
                     i++;
                 }
-            } else {
-                /* Internal thread can raise event on this stage before we
-                 * start rx processing. In this case we can return event
-                 * in right order. It is done to avoid locking and
-                 * may be it is not so critical
-                 */
+            } else if (likely(do_poll)) {
```

Comment:
may be simplify logic

```
            } else {
                mem_buf_desc_t *desc = m_p_cq_mgr_rx->poll_and_process_socketxtreme();
                if (likely(desc) && (++pkts < completions)) {
                    rx_process_buffer(desc, NULL);
                } else {
                    m_gro_mgr.flush_all(NULL);
                    break;
                }
            }
```

Reply:
we will leak desc here when first condition is true but the second is false.